cmd/copyEnumeratorInit.go

```diff
@@ -59,16 +76,87 @@ func (cca *cookedCopyCmdArgs) initEnumerator(jobPartOrder common.CopyJobPartOrde
 	// Check if the destination is a directory so we can correctly decide where our files land
 	isDestDir := cca.isDestDirectory(dst, &ctx)
 
+	// Create a S3 bucket resolver
+	// Giving it nothing to work with as new names will be added as we traverse.
+	s3BucketResolver := NewS3BucketNameToAzureResourcesResolver(make([]string, 0))
+
+	srcLevel, err := determineLocationLevel(cca.source, cca.fromTo.From(), true)
+
+	if err != nil {
+		return nil, err
+	}
+
+	dstLevel, err := determineLocationLevel(cca.destination, cca.fromTo.To(), false)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if srcLevel == ELocationLevel.Object() && dstLevel == ELocationLevel.Service() {
+		return nil, errors.New("cannot transfer files/folders to a service")
+	}
+
+	// When copying a container directly to a container, strip the top directory
+	if srcLevel == ELocationLevel.Container() && dstLevel == ELocationLevel.Container() && cca.fromTo.From() != common.ELocation.Local() && cca.fromTo.To() != common.ELocation.Local() {
+		cca.stripTopDir = true
+	}
+
+	var logDstContainerCreateFailureOnce sync.Once
 	filters := cca.initModularFilters()
 	processor := func(object storedObject) error {
-		src := cca.makeEscapedRelativePath(true, isDestDir, object)
-		dst := cca.makeEscapedRelativePath(false, isDestDir, object)
+		// Start by resolving the name and creating the container
+		if object.containerName != "" && cca.fromTo.From() != common.ELocation.Local() && cca.fromTo.To() != common.ELocation.Local() {
+			cName, err := s3BucketResolver.ResolveName(object.containerName)
+
+			if err != nil {
+				glcm.Error(err.Error())
+				return errors.New("failed to add transfers from service, some of the buckets have invalid names for the destination. " +
+					"Please exclude the invalid buckets in service to service copy, and copy them using bucket to container/share/filesystem copy " +
```

Comment:
>exclude the invalid buckets [](start = 13, length = 27)

how do they exclude buckets?

Reply:
You'd have to use a wildcard to do so. Perhaps we should just automatically exclude them and log that we excluded them.

Reply:
Yes, that sounds like a better experience, since it's hard to use wildcard for that purpose, as they have to figure out a pattern that works for all the other buckets except the one to exclude.

Reply:
Done